From e370834589952d41d32899eff860b06a8b0fd0a0 Mon Sep 17 00:00:00 2001
From: Thomas Perl <thomas.perl@jollamobile.com>
Date: Mon, 14 Oct 2013 02:51:06 +0200
Subject: [PATCH 19/65] zypp: Don't count patterns in progress calculation

Patterns are not downloaded or installed, they just refer to other
packages as dependencies to be pulled in. When calculating the number
of packages to be downloaded and installed, skip patterns.
---
 backends/zypp/pk-backend-zypp.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/backends/zypp/pk-backend-zypp.cpp b/backends/zypp/pk-backend-zypp.cpp
index ec2d9293c..9b1b9d7da 100644
--- a/backends/zypp/pk-backend-zypp.cpp
+++ b/backends/zypp/pk-backend-zypp.cpp
@@ -1606,6 +1606,11 @@ zypp_perform_execution (PkBackendJob *job, ZYpp::Ptr zypp, PerformType type, gbo
 		// Get number of installations and removals for overall progress
 		priv->exec.reset();
 		for (ResPool::const_iterator it = pool.begin (); it != pool.end (); ++it) {
+			// Patterns are not "installed" or "downloaded" as such
+			if (it->satSolvable().kind() == ResKind::pattern) {
+				continue;
+			}
+
 			if (it->status().isToBeInstalled()) {
 				if (!only_download) {
 					priv->exec.total_installs += 1;
-- 
2.13.5

